docs/changelog.rst

```diff
@@ -2,6 +2,8 @@
 Changelog
 =========
 
```

Comment:
Should we mention bug fixes here as well?

Reply:
We should. But the whole point of bug fixes is that the person who fixed the bug fix should be adding to the changelog when the bug fix happens and only if it's a user facing bug. That's what we have been doing in the past.

Can't add them all when we make a release. That is way too error-prone for the person who makes the bump.

Reply:
- I think it's inconvenient to add it with the bug fix, as that was a major reason for merge conflicts in the past. I'd propose a project or milestone for each release, so the release master can see all issues that were worked on in a quick way and can fix the changelog.
- According to your argument the node version check shouldn't be added in this PR 😜

Reply:
Makes sense. Maybe we should create a small PR checklist like:

- [ ] Code change is covered by test
- [ ] Documentation updated (if necessary)
- [ ] Change added to ChangeLog (if user visible bug fix or new feature)
...

Reply:
> I think it's inconvenient to add it with the bug fix, as that was a major reason for merge conflicts in the past.

Lol come on. Solving a conflict at the changelog is 1-2 seconds job. That's not an argument.

> According to your argument the node version check shouldn't be added in this PR

It's user facing. Users who do not use a supported version will end up getting a fat big error message. Does not get more user facing than that.

Plus, by adding changes to the changelog AT the time of bugfix we and our users get to see what's coming when checking the documentation. This is one of the reasons we use the `releases` sphinx extension:

![2019-07-30-110026_567x273_scrot](https://user-images.githubusercontent.com/1658405/62115847-67b32b80-b2b9-11e9-83bd-44e0773cb632.png)


@karlb 

> Makes sense. Maybe we should create a small PR review checklist like:

This is a dejavu. We actually do have that. Check our PR templates ;)
https://github.com/raiden-network/raiden/blob/51438f9fae326565923089346e28479f1aebf7d6/.github/PULL_REQUEST_TEMPLATE/feature.md

Reply:
> Lol come on. Solving a conflict at the changelog is 1-2 seconds job. That's not an argument.

Fixing the merge conflict is easy, but I have to stop working on the current problem, rebase, wait for the test runs again... I still believe this should be done by one person before the release.

Reply:
> Fixing the merge conflict is easy, but I have to stop working on the current problem, rebase, wait for the test runs again.

Stop working on the current problem for 1 second, rebase, push with [skip ci] since no changes were made, and then resume what you were working on.

As I said this is not an argument.

Reply:
Two things.

On the original problem of curating the changelog: I can live with the current way, I just think it it's inefficient and that's what I wanted to bring up.


> As I said this is not an argument.

On the way this discussion went: I'm irritated how you handled my arguments and didn't take them seriously. I want to be able to mention my opinions without being reprimanded by you.

Reply:
> On the original problem of curating the changelog: I can live with the current way, I just think it it's inefficient and that's what I wanted to bring up.

Good. AS you saw we disagree on that.

> On the way this discussion went: 

You are right. I should have indicated my disagreement in a less aggressive way. I apologize for that. :hugs:

Reply:
> Makes sense. Maybe we should create a small PR review checklist like:

@karlb FIY, we do have it, no one is using it though: https://github.com/raiden-network/raiden/tree/develop/.github/PULL_REQUEST_TEMPLATE

Reply:
We can just fix the problem with the changelog by using the same process from the cpython developement. They have this tool to avoid conflicts: https://pypi.org/project/blurb/